### PR TITLE
Add return type to GetRawFileOrLFS and GetRawFile

### PR DESCRIPTION
**Pull request by AdamMajer**
Document return type for the endpoints that fetch specific files from a repository. This allows the swagger generated code to read the returned data.

### CODE REVIEW

**Review by lunny — APPROVED** [review]

File: routers/api/v1/repo/file.go [hunk 1]
```diff
@@ -67,6 +67,8 @@ func GetRawFile(ctx *context.APIContext) {
 	// responses:
 	//   200:
 	//     description: Returns raw file content.
+	//     schema:
+	//       type: file
```
Comment (silverwind) [inline routers/api/v1/repo/file.go:71]: Is this a valid value for `type`? I think not, as per:

https://json-schema.org/understanding-json-schema/reference/type

Comment (AdamMajer) [reply routers/api/v1/repo/file.go:71]: It's a raw file content, not json schema. See,

https://swagger.io/docs/specification/2-0/describing-responses/#response-that-returns-a-file

**Comment by AdamMajer** [comment]
While the interface is now working and getting the file content, I seem to be getting an error

`... is not supported by the TextConsumer, can be resolved by supporting TextUnmarshaler interface"}`

I'll convert this to draft as I investigate the cause.

**Comment by AdamMajer** [comment]
I've found a potential issue here. Not with the fix but what it entails. Currently, the swagger generated code doesn't have ability to consume the Body since it's not parsed. By adding the return type to file, it's possible to pass the writer that then can consume the Body.

Swagger generated code uses returned mime-type to determine the type of consumer it allocates. So for text/plain mime, it allocates TextConsumer. Then this complains as above unless there is a trivial TextUnmarshaller() . This means that it *may be* possible to fix these weird issues by hardcoding "application/octet-stream" as the file type instead of trying to detect it in the Gitea's GetRawFile() endpoints. I'll try that tomorrow.

**Review by silverwind — APPROVED** [review]

**Comment by silverwind** [comment]
@AdamMajer If you want to wait with merging, please put PR into draft status :)

**Comment by AdamMajer** [comment]
It's OK to get this merged. The other problem here is with go-swagger and mime-type combinations. It should be addressed elsewhere.

**Review by wolfogre — APPROVED** [review]

**Pull request merged**